DavidVallejos/TicTacToe/src/com/company/Game.java

```diff
@@ -0,0 +1,166 @@
+        return true;
+    }
+
+    /**
+     * Gets current position comparing at position (i,j)
+     *
+     * @param position the position that player chose.
+     * @return boolean: true if position is was successful play, false if is invalid play
+     */
+    private boolean getPosition(int position) {
+        for (int i = 0; i < ROW; i++) {
+            for (int j = 0; j < COLUMN; j++) {
+                if (Character.getNumericValue(grid[i][j]) == position) {
+                    return playAt(i, j);
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Changes character turn
+     */
+    private char nextTurn() {
+
+        if (character == 'x') {
+            character = 'o';
+        } else {
+            character = 'x';
+        }
+
+        return character;
+    }
+
+
+    /**
+     * Checks if the game has ended either because a player has won, or if the game has ended as a tie.
+     * If game hasn't ended the return string has to be "None",
+     * If the game ends as tie, the return string has to be "Tie",
+     * If the game ends because there's a winner, it should return "X wins" or "O wins" accordingly
+     *
+     * @return String indicating the outcome of the game: "X wins" or "O wins" or "Tie" or "None"
+     */
+    private boolean checkGameWinner() {
+        return checkRow() || checkColumn() || checkDiagonals();
+    }
+
+    /**
+     * Checks if the current character win in Diagonal
+     *
+     * @return boolean: true if there is a 3 in a row, else if there is not
+     */
+    private boolean checkDiagonals() {
+        return (grid[0][0] == character && grid[1][1] == character && grid[2][2] == character) ||
+                (grid[0][2] == character && grid[1][1] == character && grid[2][0] == character);
+    }
+
+    /**
+     * Checks if the current character win in a Column
+     *
+     * @return boolean: true if there is a 3 in a row, else if there is not
+     */
+    private boolean checkColumn() {
+        for (int j = 0; j < COLUMN; j++)
+            if (grid[0][j] == character && grid[1][j] == character && grid[2][j] == character)
+                return true;
+        return false;
+    }
+
+    /**
+     * Checks if the current character win in a Row
+     *
+     * @return boolean: true if there is a 3 in a row, else if there is not
+     */
+    private boolean checkRow() {
+        for (int i = 0; i < ROW; i++)
+            if (grid[i][0] == character && grid[i][1] == character && grid[i][2] == character)
+                return true;
+        return false;
+    }
+
+    /**
+     * Starts game
+     */
+    public void playGame() {
+        Scanner option = new Scanner(System.in);
+        int value;
+        for (int turn = 1; turn <= 9; turn++) {
+            printBoard();
+            do {
+                System.out.println("Is the turn of: " + character + ", select an option between 1-9");
+                value = option.nextInt();
```

Comment:
If I enter any other type different from an int it will break the app. Can we have exception handling here? Something like TryParse(input) and fallback to keep asking for user input until a valid integer is entered?